docs: PEM-6808: Adding AWS Secret Cloud documentation (#5624)
spectrocloud/librarium: docs/docs-content/clusters/public-cloud/aws/add-aws-accounts.md

---

Palette supports integration with Amazon Web Services (AWS) Cloud Accounts, including
[AWS GovCloud (US)](https://aws.amazon.com/govcloud-us/?whats-new-ess.sort-by=item.additionalFields.postDateTime&whats-new-ess.sort-order=desc)
and [AWS Secret Cloud (US)](https://aws.amazon.com/federal/secret-cloud/) accounts. This section explains how to create
an AWS cloud account in Palette. You can use any of the following authentication methods to register your cloud account.

- AWS

  - [Static Access Credentials](#static-access-credentials)
  - [Dynamic Access Credentials](#dynamic-access-credentials)

- AWS GovCloud (US)

  - [Static Access Credentials](#static-access-credentials-1)
  - [Dynamic Access Credentials](#dynamic-access-credentials-1)

- AWS Secret Cloud (SC2S) (US)

  - [Static Access Credentials](#static-access-credentials-2)
  - [Secure Compliance Validation Credentials](#secure-compliance-validation-credentials)

## AWS Account

This section provides guidance on creating an AWS account that uses static or dynamic access credentials.

### Static Access Credentials

Use the steps below to add an AWS cloud account using static access credentials.

#### Prerequisites

> - A Palette account with [tenant admin](../../../tenant-settings/tenant-settings.md) access

Author: Not a silly question at all! So, since the bullets are not complete sentences, I removed the periods. However, when you have multiple thoughts together (even if they're not standalone sentences), I use the periods to separate them. 

I believe it is more a matter of style than anything, as I see companies using both methods. I will add back the periods if that is our preference. I'm just defaulting to what I know.

### Prerequisites

- [Palette VerteX installed](../../../vertex/install-palette-vertex/install-palette-vertex.md) and
  [tenant admin](../../../tenant-settings/tenant-settings.md) access

- The **AwsSecretPartition** [feature flag](../../../vertex/system-management/feature-flags.md) enabled in the Palette
  VerteX [system console](../../../vertex/system-management/system-management.md)

- An AWS account with an [IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user.html) or
  [IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) for Palette VerteX

- An AWS account with the [required IAM policies](required-iam-policies.md) assigned to the Palette VerteX IAM user or
  IAM role

- A secure connection to your AWS Secret Cloud account, such as via a Private Cloud Gateway (PCG) or Wide Area Network
  (WAN) tunnel

### Static Access Credentials

Use the steps below to add an AWS Secret Cloud account using static access credentials.

#### Add AWS Secret Cloud to Palette VerteX

> 1. Log in to [Palette](https://console.spectrocloud.com) as tenant admin.

Reviewer: ```suggestion
1. Log in to the Palette VerteX console as tenant admin.
```

   Toggle **Add IAM Policies** on.

7. Use the **drop-down Menu**, which lists available IAM policies in your AWS account, to select any desired IAM
   policies you want to assign to the Palette IAM role or IAM user.

8. If you are using a PCG to connect to your AWS Secret Cloud account, toggle **Connect Private Cloud Gateway** on, and
   select a **Private Cloud Gateway** from the list. This list is populated automatically with the **Private Cloud
   Gateways** listed in **Tenant Settings**. For more information, refer to the
   [Private Cloud Gateway](../../../clusters/pcg/pcg.md) page.

9. **Confirm** your AWS Secret Cloud account.

#### Validate

You can verify that the account is available in Palette by reviewing the list of cloud accounts. To review the list of
cloud accounts, navigate to the left **Main Menu**. Click on **Tenant Settings**. Next, click **Cloud Accounts**. Your
newly added AWS cloud account is listed under the AWS section.

### Secure Compliance Validation Credentials

Use the steps below to add an AWS Secret Cloud account using SCAP secure compliance validation credentials.

> #### Add AWS Secret Cloud to Palette VerteX

Reviewer: I think the validate section is missing :)

> 1. Log in to [Palette](https://console.spectrocloud.com) as Tenant admin.

Reviewer: ```suggestion
1. Log in to the Palette VerteX console as tenant admin.
```

   | **Agency Name**                    | Enter the SCAP agency name.                                                                                                                                                                                                                                                                                                                                                                                           |
   | **Account Name**                   | Enter the SCAP account name or number.                                                                                                                                                                                                                                                                                                                                                                                |

>    | **CAP/SCAP Role Name**             | Enter the role name provided by SCAP administrator. This role determines the AWS permissions granted to the account. Note that AWS Top Secret Cloud Access Portal (CAP) credentials are not supported at this time.                                                                                                                                                                                                   |

Reviewer: ```suggestion
   | **CAP/SCAP Role Name**             | Enter the role name provided by the SCAP administrator. This role determines the AWS permissions granted to the account. Note that AWS Top Secret Cloud Access Portal (CAP) credentials are not supported at this time.                                                                                                                                                                                                   |
```

   | **User Certificate**               | Paste your user-issued digital certificate in PEM-encoded format.                                                                                                                                                                                                                                                                                                                                                     |
   | **User Key**                       | Provide the private cryptographic key associated with the user certificate in PEM-encoded format.                                                                                                                                                                                                                                                                                                                     |

> 6. **Validate** the credentials.

Reviewer: ```suggestion
6. Click **Validate** to validate the credentials.
```

7. Once the credentials are validated, verified by a green check mark, the **Add IAM Policies** toggle is displayed.
   Toggle **Add IAM Policies** on.

8. Use the **drop-down Menu**, which lists available IAM policies in your AWS account, to select any desired IAM
   policies you want to assign to the Palette IAM role or IAM user.

9. If you are using a PCG to connect to your AWS Secret Cloud account, toggle **Connect Private Cloud Gateway** on, and
   select a **Private Cloud Gateway** from the list. This list is populated automatically with the **Private Cloud
   Gateways** listed in **Tenant Settings**. For more information, refer to the
   [Private Cloud Gateway](../../../clusters/pcg/pcg.md) page.

> 10. **Confirm** your AWS Secret Cloud account.

Reviewer: ```suggestion
10. Click **Confirm** to create your AWS Secret Cloud account.
```
